feat(schedule): revamp schedule display in grid

flynnhou opened the pull request:
(Thanks for sending a pull request! Please fill in the following content to let us know better about this change.)

## Types of changes
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
Based on the [wireframe](https://docs.google.com/document/d/1sxfUqtaPaaJjnVXBvcazdEXa2oQrUPnZ9VwBoo_jkN4) provided by the design team:
- Display the schedule in grid system to provide users a glimpse of overall events including break time etc.
- Tweak events data in `views.py` to help frontend display the grid layout easier
- Take the liberty temporarily to design the style of the events and the schedule

> There are still some details to better realize the ideal of the wireframe

## Steps to Test This Pull Request
Steps to reproduce the behavior:
1. Go to the `/admin` to make sure there are events created just like the ones in the existing Google sheet schedule - including Keynote events, Custom events, Talk events, Tutorial events
2. Go to the `/events/schedule/new` to review the laid out schedule
3. Click on the **Generate** button at the bottom of the `/events/schedule/new`
4. See if the new schedule is published at the `/events/schedule`

## Expected behavior
A clear and concise description of what you expected to happen.

## Related Issue
#813

## More Information
**Screenshots**

- Locate at `/events/schedule`
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/7352613/91064765-452d4680-e662-11ea-8411-106ff1e43110.png">

- Scroll down the schedule timetable
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/7352613/91064836-5ece8e00-e662-11ea-98a4-a09f4d3d5045.png">

- Hover an event (also hover at the long-line title)
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/7352613/91065195-c4227f00-e662-11ea-91ca-e4e4b7f9b36b.png">

yychen commented:
![圖片](https://user-images.githubusercontent.com/484883/91188997-7ec68600-e724-11ea-8c06-31e683d1a770.png)
我把
```css
grid-template-rows: repeat(156, 1fr);
```
槓掉就秒變 compact 耶!!!! !!!

yychen commented:
I think we can have a toggle option fixed on the page, like, at the upper-left corner. We let the user toggle on / off the normal & compact mode. Compact mode actually looks nice to me.

yychen commented on src/static/pycontw-2020/styles/pages/_events.scss:
```diff
@@ -0,0 +1,300 @@
+						.context {
+							.title {
+								margin-bottom: 8px;
+								font-size: 22px;
```
I think the font is too large

yychen commented on src/static/pycontw-2020/styles/pages/_events.scss:
```diff
@@ -0,0 +1,300 @@
+			&__tab {
+				margin: 0 24px 24px;
+				padding: 24px 72px;
+				font-family: Philosopher, sans-serif;
```
Do you consider to use something like `$header-font-family` instead?

yychen commented on src/static/pycontw-2020/styles/pages/_events.scss:
```diff
@@ -0,0 +1,300 @@
+					margin: 8px 8px 8px 0;
+					padding: 12px 0;
+					justify-content: center;
+					font-family: Quicksand, sans-serif;
```
This one with `$text-font-family`?

yychen commented on src/static/pycontw-2020/styles/pages/_events.scss:
```diff
@@ -0,0 +1,300 @@
+
+				&-item {
+					display: flex;
+					font-family: Quicksand, sans-serif;
```
This one also

yychen commented on src/static/pycontw-2020/styles/pages/_events.scss:
```diff
@@ -0,0 +1,300 @@
+				.timeline {
+					grid-column: times;
+					justify-self: center;
+					font-family: Philosopher, sans-serif;
```
This one also

yychen commented on src/templates/pycontw-2020/_includes/menu.html:
```diff
@@ -6,7 +6,7 @@
 {% url 'page' path='about/staff' as about_staff_url %}
 {% url 'page' path='conference/overview' as events_overview_url %}
-{% url 'events_schedule' as events_schedule_url %}
+{% url 'page' path='events/schedule' as events_schedule_url %}
```
Why change to this?

yychen commented on src/static/pycontw-2020/styles/pages/_events.scss:
```diff
@@ -0,0 +1,300 @@
+						@include icon('room-R3', 1px, center);
+					}
+
+					.grass {
+						@include icon('grass-group', 0, center);
+
+						&::before {
+							background-size: contain;
+						}
+					}
+
+					.snake {
+						@include icon('snake', 0, center);
+
+						&::before {
+							background-size: contain;
+						}
+					}
```
Is it possible that we change it to this year's theme? like the window pattern or the ancient tower (赤崁樓) icon? The grass and snake is kind of the Pokemon's theme for last year.

yychen commented:
And, I think you'll also need to change the `src/events/models.py`, hack the names into the following:
```py
    LOCATION_CHOICES = [
        (Location.ALL,  _('All rooms')),
        # (Location.R012, _('R0, R1, R2')),
        # (Location.R0,   _('R0')),
        # (Location.R1,   _('R1')),
        # (Location.R2,   _('R2')),
        # (Location.R3,   _('R3')),
        # (Location.R4,   _('R4')),

        (Location.R012, _('R1, R2, R3')),
        (Location.R0,   _('R1')),
        (Location.R1,   _('R2')),
        (Location.R2,   _('R3')),
        (Location.R3,   _('Multifunction room')),
        (Location.R4,   _('Goodideas Studio')),
    ]
```

flynnhou commented:
@yychen based on your comments above, I've updated the UI and some refactors.
- In short, the UI would look like this with a new simple button to toggle the timetable's grid mode between `default` or `compact`.

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/7352613/91230154-9706c700-e75d-11ea-9a91-66ae325af441.png">

- The rest of the updates please check out the commits I'd pushed 🙏. I'm too lazy to reply to each of your comment 😛

flynnhou commented:
@yychen I've rebased the master branch thus the forced push.

Based on the offline discussions regarding the style of the schedule, I've modified the below changes:
(1) Remove the unnecessary icon decoration of the custom events - break, lunch, etc.
(2) Set the grid to compact mode by default for it's unnecessary to reflect the actual time length of an event
(3) Take out the text decoration on `<a>`
(4) Change the events with better contrast background colors

![photo_2020-08-28_01-13-42](https://user-images.githubusercontent.com/7352613/91477557-18cb3180-e8d1-11ea-8be5-de92026ee5ad.jpg)

flynnhou commented:
- Improved desktop schedule styles
<img width="721" alt="image" src="https://user-images.githubusercontent.com/7352613/91613358-ef360700-e9b1-11ea-8742-c24e7a9bfa75.png">

- Added mobile view schedule
<img width="449" alt="image" src="https://user-images.githubusercontent.com/7352613/91613446-20163c00-e9b2-11ea-8f44-51258554588c.png">
<img width="450" alt="image" src="https://user-images.githubusercontent.com/7352613/91613489-30c6b200-e9b2-11ea-85d2-d7f792198f28.png">

yychen reviewed (approved):
They look very nice! Thank you for your hard work.
